app/main.py

```diff
@@ -1,8 +1,32 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
+
+    @classmethod
+    def add_spouse(cls, person: dict, spouse_name: str) -> None:
+        spouse = cls.people.get(spouse_name)
+        if spouse:
+            if hasattr(person, "wife"):
+                person.wife = spouse
+                spouse.husband = person
+            else:
+                person.husband = spouse
+                spouse.wife = person
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_data in people:
+        name = person_data["name"]
+        age = person_data["age"]
+        person = Person(name, age)
+        person_list.append(person)
+        spouse_key = "wife" if "wife" in person_data else "husband"
+        spouse_name = person_data[spouse_key]
+        if spouse_name:
+            Person.add_spouse(person, spouse_name)
```

Comment:
all this logic is overcomplicated. You can follow this to use less if statements:
if wife in person data -> set person wife to an object that corresponds to wife name + set wifes husband as current person.
repeat same logic for husband.